acceptance-tests/src/main/kotlin/org/ostelco/at/jersey/Tests.kt

```diff
@@ -1897,7 +1899,157 @@ class PlanTest {
             purchaseRecords.sortBy { it.timestamp }
 
             assert(Instant.now().toEpochMilli() - purchaseRecords.last().timestamp < 10_000) { "Missing Purchase Record" }
-            assertEquals(expectedPlanProduct, purchaseRecords.last().product, "Incorrect 'Product' in purchase record")
+            assertEquals(expectedPlanProductSG, purchaseRecords.last().product, "Incorrect 'Product' in purchase record")
+        } finally {
+            StripePayment.deleteCustomer(customerId = customerId)
+        }
+    }
+}
+
+class RenewPlanTest {
+
+    @Test
+    fun `jersey test - POST purchase plan with trial time`() {
+
+        val email = "purchase-${randomInt()}@test.com"
+        val sku = "PLAN_10USD_DAY"
+        var customerId = ""
+
+        try {
+            customerId = createCustomer(name = "Test Purchase Plan User", email = email).id
+            enableRegion(email = email, region = "us")
+
+            val sourceId = StripePayment.createPaymentTokenId()
+
+            post<String> {
+                path = "/products/$sku/purchase"
+                this.email = email
+                queryParams = mapOf("sourceId" to sourceId)
+            }
+
+            Thread.sleep(200) // wait for 200 ms for balance to be updated in db
```

Comment:
This is brittle.   Is there a less brittle way to wait for the database?

Reply:
Not sure. The same `wait` statement is present in all acceptance tests that involves updates to `neo4j`. Will add a TODO and a Notion tech dept task.